src/js/modules/infragistics.ui.tree.js

```diff
@@ -3678,155 +3804,114 @@
-								checkbox = parentLi.children("span[data-role=checkbox]");
-								checkIcon = checkbox.children("span");
-								checkbox.attr("data-chk", "off");
-								checkIcon.removeClass(css.checkboxOn).removeClass(css.checkboxPartial)
-									.addClass(css.checkboxOff);
-								parentLi = this.parentNode(parentLi);
-							} else {
-								checkbox = parentLi.children("span[data-role=checkbox]");
-								checkIcon = checkbox.children("span");
-								checkbox.attr("data-chk", "partial");
-								checkIcon.removeClass(css.checkboxOn).removeClass(css.checkboxOff)
-									.addClass(css.checkboxPartial);
-								parentLi = this.parentNode(parentLi);
-							}
-						}
-					} else {
-						childCheckboxes = node.find("span[data-role=checkbox]");
-						childCheckIcons = childCheckboxes.children("span");
-						checkbox.attr("data-chk", "on");
-						checkIcon.removeClass(css.checkboxOff).removeClass(css.checkboxPartial)
-							.addClass(css.checkboxOn);
-						childCheckboxes.attr("data-chk", "on");
-						childCheckIcons.removeClass(css.checkboxOff).removeClass(css.checkboxPartial)
-							.addClass(css.checkboxOn);
-
-						parentLi = checkbox.parent().parent().parent();
-						while (parentLi && parentLi.is("li")) {
-
-							// All childcheckboxes are checked
-							if (parentLi.find("ul > li > span[data-chk=on]").length ===
-									parentLi.find("ul > li").length) {
-								checkbox = parentLi.children("span[data-role=checkbox]");
-								checkIcon = checkbox.children("span");
-								checkbox.attr("data-chk", "on");
-								checkIcon.removeClass(css.checkboxOff).removeClass(css.checkboxPartial)
-									.addClass(css.checkboxOn);
-								parentLi = this.parentNode(parentLi);
-							} else {
-								checkbox = parentLi.children("span[data-role=checkbox]");
-								checkIcon = checkbox.children("span");
-								checkbox.attr("data-chk", "partial");
-								checkIcon.removeClass(css.checkboxOn).removeClass(css.checkboxOff)
-									.addClass(css.checkboxPartial);
-								parentLi = this.parentNode(parentLi);
-							}
-						}
-					}
+				state = nodeObject.data[ nodeObject.binding.checkedKey ];
+				if (state === "on" || state === "partial") {
+					this.uncheckNode(nodeObject);
 				} else {
-					if (checkbox.attr("data-chk") === "on" || checkbox.attr("data-chk") === "partial") {
-						checkbox.attr("data-chk", "off");
-						checkIcon.removeClass(css.checkboxOn).removeClass(css.checkboxPartial)
-							.addClass(css.checkboxOff);
-					} else {
-						checkbox.attr("data-chk", "on");
-						checkIcon.removeClass(css.checkboxOff).addClass(css.checkboxOn);
-					}
+					this.checkNode(nodeObject);
+				}
+				if (event) {
+					this._triggerNodeCheckstateChanged(event, nodeObject);
 				}
-				self._triggerNodeCheckstateChanged(event, node);
 			}
 		},
-		toggle: function (node, event) {
-			/* Toggles the collapse/expand state for the specified node.
+		checkNode: function (nodeObj, cascadeDir) {
```

Comment:
Is this intended to a new public method? If so, the `nodeObj` param is very tricky to create for a mere mortal - 
the format is not really described and the `_constructNodeObject` is kinda private (events expose node objects, but that's not reasonable to go for).
This should probably be folded under a `_checkNode` with the public method accepting a DOM elements as the "node element" description implies based on other methods.
This also applies to `uncheckNode` and `partiallyCheckNode` ofc.

Reply:
The public API that should be used to get the node objects is [`nodeFromElement`](https://www.igniteui.com/help/api/2017.1/ui.igtree#methods:nodeFromElement)

Reply:
Right, missed there's a public method that does the exact same thing :)